vulnerabilities/views.py

```diff
@@ -19,50 +19,118 @@
+                    context["package_form"] = PackageForm(request.GET or None)
+                    context["template_name"] = "packages.html"
+                    return render(request, "packages.html", context)
+                elif request.GET.get("template") == "package_details":
+                    context["package_form"] = PackageForm(request.GET or None)
+                    context["template_name"] = "package_update.html"
+                    return render(request, "package_update.html", context)
+                elif request.GET.get("template") == "index":
+                    context["package_form"] = PackageForm(request.GET or None)
+                    context["vuln_form"] = CVEForm(request.GET or None)
+                    context["template_name"] = "index.html"
+                    return render(request, "index.html", context)
+                else:
+                    context["package_form"] = PackageForm(request.GET or None)
+                    context["vuln_form"] = CVEForm(request.GET or None)
+                    context["template_name"] = "index.html"
+                    return render(request, "index.html", context)
+
+        context["package_form"] = PackageForm(request.GET or None)
+        context["template_name"] = self.template_name
         return render(request, self.template_name, context)
 
     @staticmethod
     def request_to_queryset(request):
         package_type = ""
         package_name = ""
+        purl = ""
 
         if len(request.GET["type"]):
             package_type = request.GET["type"]
 
         if len(request.GET["name"]):
             package_name = request.GET["name"]
 
+        # Check whether the input value is a syntactically-correct purl
+        try:
+            purl = PackageURL.from_string(package_name)
+            return list(
+                models.Package.objects.all()
+                .filter(Q(type=purl.type, name=purl.name, version=purl.version))
+                .annotate(
+                    vulnerability_count=Count(
+                        "vulnerabilities",
+                        filter=Q(packagerelatedvulnerability__fix=False),
+                    ),
+                    # TODO: consider renaming to fixed in the future
+                    patched_vulnerability_count=Count(
+                        "vulnerabilities",
+                        filter=Q(packagerelatedvulnerability__fix=True),
+                    ),
+                )
+                .prefetch_related()
+            )
+        except:
+            pass
+
         return list(
             models.Package.objects.all()
-            # FIXME: This filter is wrong and ignoring most of the fields needed for a
-            # proper package lookup: type/namespace/name@version?qualifiers and so on
             .filter(name__icontains=package_name, type__icontains=package_type)
+            .order_by("type", "namespace", "name", "version", "subpath", "qualifiers")
```

Comment:
This is also a good candidate to move on the PackageQuerySet class:

```
def order_by_purl(self):
    self.order_by("type", "namespace", "name", "version", "subpath", "qualifiers")
```

Then on the view side:

```
models.Package.objects.[...].order_by_purl()
```

Reply:
See new issue https://github.com/nexB/vulnerablecode/issues/880